### ISSUE TITLE
Various compilation errors when schema, table, key names conflict

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10561]
Generating code for this schema produces various compilation errors in `Indexes.java` and `Keys.java` for both `b_c` and `bc` schemata:

```sql
CREATE SCHEMA a;
CREATE SCHEMA b_c;

CREATE TABLE a.a (a int, CONSTRAINT a PRIMARY KEY (a));
CREATE TABLE a.b_c (b_c int, CONSTRAINT b_c PRIMARY KEY (b_c));

CREATE INDEX a.a ON a.a (a);
CREATE INDEX a.b_c ON a.b_c (b_c);

CREATE TABLE b_c.a (a int, CONSTRAINT a PRIMARY KEY (a));
CREATE TABLE b_c.b_c (b_c int, CONSTRAINT b_c PRIMARY KEY (b_c));

CREATE INDEX b_c.a ON b_c.a (a);
CREATE INDEX b_c.b_c ON b_c.b_c (b_c);
```

As a workaround, as always, a generator strategy can be implemented.

**Comment by lukaseder** [comment]
Related: #8808. The solution there can be generalised.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.14.0. Backport scheduled for 3.13.5 (#10562)
